src/PageHeader/PageHeader.stories.tsx

```diff
@@ -0,0 +1,413 @@
+      if: {arg: 'ContextArea.hidden', truthy: false},
+      table: {
+        category: 'ContextArea Slot',
+        defaultValue: {
+          summary: `
+        {
+          narrow: false,
+          regular: true,
+          wide: true
+        }
+        `
+        }
+      },
+      description: 'Parent '
+    },
+    'ContextBar.hidden': {
+      type: 'boolean',
+      if: {arg: 'ContextArea.hidden', truthy: false},
+      table: {
+        category: 'ContextArea Slot',
+        defaultValue: {
+          summary: `
+        {
+          narrow: false,
+          regular: true,
+          wide: true
+        }
+        `
+        }
+      },
+      description:
+        'ContextBar is generic slot for any component above the title region. Use it for custom breadcrumbs and other navigation elements instead of ParentLink.'
+    },
+    'ContextAreaAction.hidden': {
+      type: 'boolean',
+      if: {arg: 'ContextArea.hidden', truthy: false},
+      table: {
+        category: 'ContextArea Slot',
+        defaultValue: {
+          summary: `
+        {
+          narrow: false,
+          regular: true,
+          wide: true
+        }
+        `
+        }
+      }
+    },
+    'BackButton.hidden': {
+      type: 'boolean',
+      table: {
+        category: 'TitleArea Slot',
+        defaultValue: {
+          summary: `
+        {
+          narrow: true,
+          regular: false,
+          wide: false
+        }
+        `
+        }
+      },
+      description:
+        'A back button can be used as a leading action for local navigation. On Narrow viewports, use parentLink instead.'
+    },
+    Title: {
+      type: 'string',
+      table: {
+        category: 'TitleArea Slot'
+      }
+    },
+    'Title.as': {
+      control: {
+        type: 'select'
+      },
+      options: ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'],
+      table: {
+        category: 'TitleArea Slot'
+      }
+    },
+    'Title.variant': {
+      control: {
+        type: 'radio'
+      },
+      options: ['large', 'medium', 'subtitle'],
+      table: {
+        category: 'TitleArea Slot',
+        defaultValue: {
+          summary: `
+          {
+            narrow: 'medium',
+            regular: 'large',
+            wide: 'large'
+          } `
+        }
+      },
+      description:
+        '`medium` is the most common page title size. Use for static titles in most situations. `large` for for user-generated content such as issues, pull requests, or discussions. `subtitle` when a PageHeader.Title is already present in the page, such as in a SplitPageLayout.'
+    },
+    LeadingVisual: {
+      ...OcticonArgType([CodeIcon, GitPullRequestIcon, PeopleIcon]),
+      table: {
+        category: 'TitleArea Slot'
+      },
+      description:
+        'Leading visualLeading visuals are optional and appear at the start of the title. They can be octicons, avatars, and other custom visuals that fit a small area.'
+    },
+    'LeadingVisual.hidden': {
```

Comment:
In these stories, removing these additional `LeadingVisual`, `TrailingVisual`, `TrailingAction` options requires the user to "turn on" the hiding. I wonder if we can provide a better experience.

For example, in the `SplitPageLayout` story, I've included `hasPane`, `hasHeader` toggles as the first item of each category to make it easier to parse all the functionality:

Link: https://primer.style/css/storybook/?path=/story/components-layout-beta-splitpagelayout--playground

Reply:
This is something I wasn't sure either. Thanks for providing feedback on that 🙇🏼 I was looking from a perspective that slots come visible as default and consumers can manage their visibility by hiding them. I even got confused myself when writing them and I am glad you mentioned it too.

I changed the prop from being `hidden` to `visible` now and controls became `hasContextArea`, `hasParentLink`, `hasLeadingVisual` etc and I feel it is clearer now. What you think about the current structure now?

Reply:
@mperrotti could I please get your thoughts on that? 🙏🏼

Reply:
I'm good with the `hasContextArea`, `hasParentLink`, `hasLeadingVisual` etc approach